mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -125,6 +128,52 @@ static Value getAccumulator(Value a, Value b, Value c, OpBuilder &builder,
   }
   return c;
 }
+
+static LogicalResult
+setPrefillForSplitK(func::FuncOp &func,
+                    const BufferDependencyAnalysis &bufferDeps) {
+  auto walkRes = func.walk([&bufferDeps](Operation *op) -> WalkResult {
+    Value matC;
+    int64_t splitKFactor = 1;
+    if (auto gemm = dyn_cast<GridwiseGemmAccelOp>(op)) {
+      matC = gemm.getC();
+      splitKFactor = gemm.getParams().getSplitKFactor();
+    } else if (auto gemm = dyn_cast<GridwiseGemmOp>(op)) {
+      matC = gemm.getC();
+      splitKFactor = gemm.getParams().getSplitKFactor();
+    }
+
+    if (matC && splitKFactor > 1) {
+      auto func = llvm::cast<func::FuncOp>(op->getParentOp());
+      OpBuilder builder(func->getContext());
+      FailureOr<SmallVector<BlockArgument>> args =
+          traceGemmOutputToArgs(matC, func, builder, bufferDeps);
+      if (failed(args)) {
+        op->emitError("can't trace gemm output to output argument");
+        return WalkResult::interrupt();
+      }
+
+      // initialize to zeros
+      auto elementType = cast<MemRefType>(matC.getType()).getElementType();
+      Attribute zero;
+      if (llvm::isa<FloatType>(elementType)) {
+        zero = builder.getFloatAttr(elementType, 0.0);
+      } else {
+        assert(llvm::isa<IntegerType>(elementType) &&
```

Comment:
I d prefer a pass failure here... 
I know that we expect either int or float here but technically its provided by the user.
so it might be better to give feedback as a compilation failure rather than a crash.